accord-core/src/main/java/accord/coordinate/CheckOn.java

```diff
@@ -206,8 +208,9 @@ public Void apply(SafeCommandStore safeStore)
                         break;
 
                 case PreAccepted:
-                    if (!safeStore.ranges().at(txnId.epoch()).isEmpty())
-                        Commands.preaccept(safeStore, txnId, partialTxn, maxRoute, progressKey);
+                    // only preaccept if we coordinate the transaction
+                    if (untilLocalEpoch <= txnId.epoch() || safeStore.ranges().coordinates(txnId).intersects(maxRoute))
```

Comment:
I'm not understanding the purpose of this change. I think this would only be true (and the other side of the || not) would be if we learned of a txnId < our current epoch

Reply:
we can Accept commands that aren't owned by us for either PreAccept OR Execute, and in these cases we may know of a transaction. Also, we may learn of a transaction that we adopt a dependency on (and need to see the execution of) but also never actually intersects with us for coordination or execution. I think this case might be able to happen simply for transactions we don't coordinate but do execute, though (maybe not). Plus no doubt other weird things I'm not thinking of.